pkg/secrets/pushtofile/secrets_groups_test.go

```diff
@@ -0,0 +1,148 @@
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		assert.Equal(t, expected, result)
+	}
+}
+
+func assertProperError(expectedErr string) assertFunc {
+	return func(t *testing.T, result SecretsGroups, err error) {
+		assert.Nil(t, result)
+		assert.Contains(t, err.Error(), expectedErr)
+	}
+}
+
+const someValidSecrets = "- test/url\n- test-password: test/password\n- test-username: test/username\n"
+const someValidTemplate = "\"cache\": {\n\t\"url\": {{ .url }},\n\t\"admin-password\": {{ index . \"test-password\" }},\n\t\"admin-username\": {{ index . \"test-username\" }},\n\t\"port\": 123456,\n}"
+
+var extractSecretsGroupsTestCases = []extractSecretsGroupsTestCase{
+	{
+		description: "valid annotations map",
+		contents: map[string]string{
+			"conjur.org/authn-identity":                   "host/conjur/authn-k8s/cluster/apps/inventory-api",
+			"conjur.org/container-mode":                   "application",
+			"conjur.org/secrets-destination":              "file",
+			"conjur.org/k8s-secrets":                      "- k8s-secret-1\n- k8s-secret-2\n",
+			"conjur.org/retry-count-limit":                "10",
+			"conjur.org/retry-interval-sec":               "5",
+			"conjur.org/debug-logging":                    "true",
+			"conjur.org/conjur-secrets.cache":             someValidSecrets,
+			"conjur.org/secret-file-path.cache":           "this/relative/path",
+			"conjur.org/conjur-secrets-policy-path.cache": "some/policy/path",
+		},
+		assert: assertExpectedSecretsGroups(
+			SecretsGroups{
+				"cache": {
+					Secrets: SecretsPaths{
+						"test-password": "/some/policy/path/test/password",
+						"test-username": "/some/policy/path/test/username",
+						"url":           "/some/policy/path/test/url",
+					},
+					FilePath:     "/conjur/secrets/this/relative/path",
+					FileFormat:   SecretsFileFormat(FILE_FORMAT_YAML),
+					FilePerms:    DEFAULT_FILE_PERMS,
+					FileTemplate: "",
+				},
+			},
+		),
+	},
+	{
+		description: "invalid file path",
+		contents: map[string]string{
+			"conjur.org/conjur-secrets.cache":   someValidSecrets,
+			"conjur.org/secret-file-path.cache": "/this/absolute/path",
+		},
+		assert: assertProperError("must contain relative path"),
+	},
+	{
+		description: "invalid file path with template specified",
+		contents: map[string]string{
+			"conjur.org/conjur-secrets.cache":       someValidSecrets,
+			"conjur.org/secret-file-path.cache":     "this/relative/directory/",
+			"conjur.org/secret-file-template.cache": someValidTemplate,
+		},
+		assert: assertProperError("contains directory instead of file"),
+	},
+}
+
+func TestExtractSecretsGroupsFromAnnotations(t *testing.T) {
+	for _, tc := range extractSecretsGroupsTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			secretGroups, err := ExtractSecretsGroupsFromAnnotations(tc.contents)
+
+			tc.assert(t, secretGroups, err)
+		})
+	}
+}
+
+func TestExtractFileFormatFromAnnotations(t *testing.T) {
```

Comment:
I'm not sure the type SecretsFileFormat needs to be public. If not then we can avoid unit testing it. It is indirectly tested as part of the tests for `ExtractSecretsGroupsFromAnnotations`. 

Supposing that we keep `TestExtractFileFormatFromAnnotations`, it's currently being tested by exercising `ExtractSecretsGroupsFromAnnotations`. I think `parseFileFormat` might be the more appropriate unit of logic to test here.